File: hw/codena/u3/hw23AbstractFactoryCode/Javascript/widgets/model/LinuxFactory.js
import GUIFactory from "./GUIFactory.js";
import LinuxButton from "./LinuxButton.js";
import LinuxMenu from "./LinuxMenu.js";

export default class LinuxFactory extends GUIFactory {

    createButton() {
        return new LinuxButton();
    }

    createMenu() {
        return new LinuxMenu();
    }

}

File: hw/palomo/u3/hw23AbstracFactory/AbstracFactory.py/ec_edu_espe/model/linux_factory.py
from ec_edu_espe.model.gui_factory import GUIFactory
from ec_edu_espe.model.linux_button import LinuxButton
from ec_edu_espe.model.linux_menu import LinuxMenu


class LinuxFactory(GUIFactory):
    """ConcreteFactory: crea la familia de productos con estilo Linux."""

    def create_button(self) -> LinuxButton:
        return LinuxButton()

    def create_menu(self) -> LinuxMenu:
        return LinuxMenu()